### PR TITLE
Fix Connect for Intel Macs

### LINKED ISSUE #44320: Teleport Connect 16.1.0 doesn't start on Intel Macs
Attempting to start Teleport Connect 16.1.0 on an Intel Mac results in the following error:

```
Could not initialize tsh daemon client in the main process: Error: Could not communicate with the shared helper process.

Last logs from /Users/rav/Library/Application Support/Teleport Connect/logs/shared.log:
'/Applications/Teleport Connect.app/Contents/Resources/app.asar/node_modules/node-pty/lib/index.js',
'/Applications/Teleport Connect.app/Contents/Resources/app.asar/build/app/main/sharedProcess.js'
]
}
/Applications/Teleport Connect.app/Contents/Resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:35
throw outerError;
^

Error: dlopen(/Applications/Teleport Connect.app/Contents/Resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node, 1): no suitable image found.  Did find:
/Applications/Teleport Connect.app/Contents/Resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node: mach-o, but wrong architecture
/Applications/Teleport Connect.app/Contents/Resources/app.asar.unpacked/node_modules/node-pty/build/Release/pty.node: mach-o, but wrong architecture
at process.func [as dlopen] (node:electron/js2c/node_init:2:2559)
at Module._extensions..node (node:internal/modules/cjs/loader:1470:18)
at Object.func [as .node] (node:electron/js2c/node_init:2:2786)
at Module.load (node:internal/modules/cjs/loader:1215:32)
at Module._load (node:internal/modules/cjs/loader:1031:12)
at c._load (node:electron/js2c/node_init:2:17025)
at Module.require (node:internal/modules/cjs/loader:1240:19)
at require (node:internal/modules/helpers:179:18)
at Object.<anonymous> (/Applications/Teleport Connect.app/Contents/Resources/app.asar/node_modules/node-pty/lib/unixTerminal.js:24:11)
at Module._compile (node:internal/modules/cjs/loader:1373:14) {
code: 'ERR_DLOPEN_FAILED'
}

Node.js v20.14.0
```

I suppose that's related to the recent electron-builder update that was released in 16.1.0 (#43945).

## Further node-pty problems

Edit: Those turned out to work on other laptops, it might be an issue with my old MBP.

So I went to check v16.0.4. It boots up, but it's not possible to open a new terminal because the shared process crashes with SIGSEGV. The last logs from the shared process are just this:

```
[17-07-24 11:19:58] {CONNECT_GRPC_PORT: 1}
[17-07-24 11:20:04] [PtyHostService] info: created PTY process for id pxImejd1UUbHs72BlYs75
```

This has been broken since 15.1.10 which updated both node-pty (#39657) and electron-builder (#38964). I guess the fault here is on node-pty's side.

### PR DESCRIPTION
**Pull request by gzdunek**
Fixes https://github.com/gravitational/teleport/issues/44320
After the recent electron-builder update, Connect on amd64 Macs no longer starts because it can't find `node-pty`.
e counterpart https://github.com/gravitational/teleport.e/pull/4695

This happens because the "legacy builder" in electron-builder stopped rebuilding `node-pty` (likely a bug). I didn't find any workaround for this, so I decided to switch to the new default builder (`@electron/rebuild`). 

This one, on the other hand, has its own problem: it puts amd64 and am64 `node-pty` files into both temporary amd64 and arm64 directories (that are then combined into an universal app). `lipo` tool is not fine with it and fails with the following error `[temp-amd64/node-pty.node] and [temp-arm64/node-pty.node] have the same architectures (arm64) and can't be in the same fat output file`.

I realized that this can be solved by passing `disablePreGypCopy: true` to `@electron/rebuild` options. 
[Until it's fixed in `electron-builder`](https://github.com/electron-userland/electron-builder/issues/8347), we will use our own patch.

I also updated electron-builder to the stable version. I tested it quickly on arm64 and amd64 Mac, Windows and Ubuntu. 
Tag build: 16.0.0-dev.gzdunek.11

Changelog: Fixed a regression that caused Teleport Connect to fail to start on Intel Macs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,8 @@
     "prettier-check": "yarn prettier --check '+(e|web)/**/*.{ts,tsx,js,jsx,mts}'",
     "prettier-write": "yarn prettier --write --log-level silent '+(e|web)/**/*.{ts,tsx,js,jsx,mts}'",
     "process-icons": "node web/packages/design/src/Icon/script/script.js & yarn prettier --loglevel silent --write 'web/packages/design/src/Icon/Icons/*.tsx'",
-    "nop": "exit 0"
+    "nop": "exit 0",
+    "postinstall": "patch-package"
```
Comment (ravicious) [inline package.json:28]: The pnpm PR will have to account for this as well, right? https://pnpm.io/cli/patch

**Review by ravicious — APPROVED** [review]
It works fine for me when I run `yarn package-term --universal` locally.

File: package.json (hunk 1 above)
Comment (gzdunek) [reply package.json:28]: Correct. From what I understand, we will simply remove `patch-package` as `pnpm` supports patching out of the box.

Comment (ravicious) [reply package.json:28]: It looks like pnpm needs some entries in package.json for patches to work though. https://youtu.be/0GjLqRGRbcY?t=137

**Review by zmb3 — APPROVED** [review]

**Comment by gzdunek** [comment]
`electron-builder@25.0.1` has the fix we need, I will remove our patch.

**Pull request merged**